src/main/java/core/basesyntax/service/impl/FruitTransactionProcessor.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Storage;
+import core.basesyntax.model.impl.FruitTransaction;
+import core.basesyntax.service.TransactionProcessor;
+
+public class FruitTransactionProcessor implements TransactionProcessor<FruitTransaction> {
+    private final Storage storage;
+
+    public FruitTransactionProcessor(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void process(FruitTransaction transaction) {
+        storage.add(transaction.getProductType(), strategy(transaction));
+    }
+
+    private int strategy(FruitTransaction transaction) {
+        return switch (transaction.getTransactionType()) {
+            case PURCHASE -> subtraction(transaction);
+            case RETURN, BALANCE, SUPPLY -> add(transaction);
+        };
+    }

```

Comment:
we should replace it with the strategy map of `<OperationType, OperationHandler>` check the theory and the example that Bohdan did and why is it better than the switch
Also that `OperationHandler` can just directly access the storage instead of returning something